### PR TITLE
Reorganize Main()

### PR DESCRIPTION
**Pull request by rgayon**
Remove logic from Main() and into sub functions.
DetectStorage() sets the storage_object attribute, and raises if an issue happens (instead of exiting)
ShowHistory() is renamed to GetHistory(), which returns a str instead of displaying it

### CODE REVIEW

File: docker_explorer/de.py [hunk 1]
```diff
@@ -85,15 +82,24 @@ def DetectStorage(self):
     if os.path.isfile(
         os.path.join(self.docker_directory, 'repositories-aufs')):
       # Handles Docker engine storage versions 1.9 and below.
-      return aufs.AufsStorage(
+      self.storage_object = aufs.AufsStorage(
           docker_directory=self.docker_directory, docker_version=1)
     elif os.path.isdir(os.path.join(self.docker_directory, u'overlay2')):
```
Comment (tomchop) [inline docker_explorer/de.py]: remove the u'' here

File: docker_explorer/de.py [hunk 2]
```diff
@@ -85,15 +82,24 @@ def DetectStorage(self):
...
+          docker_directory=self.docker_directory)
     elif os.path.isdir(os.path.join(self.docker_directory, 'aufs')):
-      return aufs.AufsStorage(docker_directory=self.docker_directory)
-    return None
+      self.storage_object = aufs.AufsStorage(
+          docker_directory=self.docker_directory)
+    if self.storage_object is None:
+      err_message = (
+          'Could not detect storage system. '
+          'Make sure the docker directory ({0:s}) is correct. '
```
Comment (tomchop) [inline docker_explorer/de.py]: Maybe use a more explicit warning? "make sure the docker directory () contains one of [repositories-aufs, overlay, overlay2, aufs]".

File: docker_explorer/de.py [hunk 3]
```diff
@@ -178,6 +184,44 @@ def ParseOptions(self, options):
...
+    """
+    if self.storage_object is None:
+      self.DetectStorage()
+    self.storage_object.Mount(container_id, mountpoint)
+
+  def ShowContainers(self, only_running=False):
+    """Displays the running containers.
+
+    Args:
+      only_running(bool): Whether we display only running Containers.
```
Comment (tomchop) [inline docker_explorer/de.py]: `only_running(bool)` has no space before the variable type, whereas other docstrings do. Settle on one?

File: docker_explorer/de.py [hunk 4]
```diff
@@ -178,6 +184,44 @@ def ParseOptions(self, options):
...
+    print(self.storage_object.ShowContainers(only_running=only_running))
+
+  def ShowRepositories(self):
+    """Displays information about the images in the Docker repository."""
+    if self.storage_object is None:
+      self.DetectStorage()
+    print(self.storage_object.ShowRepositories())
+
+  def ShowHistory(self, container_id, show_empty_layers=False):
+    """Prints the history of the modifications of a container.
```
Comment (tomchop) [inline docker_explorer/de.py]: "the modification history of a container"?

File: docker_explorer/lib/storage.py [hunk 5]
```diff
@@ -328,33 +328,39 @@ def Mount(self, container_id, mount_dir):
         # TODO(romaing) this is quite unsafe, need to properly split args
         subprocess.call(c, shell=True)
 
-  def ShowHistory(self, container_id, show_empty_layers=False):
-    """Prints the history of the modifications of a container.
+  def GetHistory(self, container_id, show_empty_layers=False):
+    """Returns a string representing the history of the modifications of a
```
Comment (tomchop) [inline docker_explorer/lib/storage.py]: "representing the modification history of a container"

File: docker_explorer/de.py (hunk 1 above)
Comment (rgayon) [reply docker_explorer/de.py]: done

File: docker_explorer/de.py (hunk 2 above)
Comment (rgayon) [reply docker_explorer/de.py]: I'll totally change how storage is detected in a following PR =)

File: docker_explorer/de.py (hunk 3 above)
Comment (rgayon) [reply docker_explorer/de.py]: done

**Comment by rgayon** [comment]
Fixed all the things!

**Pull request merged**